modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/InstrumentControllerSim.scala

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.seqexec.server
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import edu.gemini.seqexec.model.dhs.ImageFileId
+import edu.gemini.seqexec.server.SeqexecFailure.SeqexecException
+import gov.aps.jca.TimeoutException
+import org.log4s.getLogger
+import squants.Time
+
+import scala.annotation.tailrec
+import scalaz.{EitherT, Show}
+import scalaz.concurrent.Task
+import scalaz.syntax.show._
+import scalaz.syntax.std.boolean._
+
+class InstrumentControllerSim(name: String) {
+  private val Log = getLogger
+
+  private val stopFlag = new AtomicBoolean(false)
+  private val abortFlag = new AtomicBoolean(false)
+  private val pauseFlag = new AtomicBoolean(false)
+
+  private val tic = 200
+
+  @tailrec
+  private def observeTic(stop: Boolean, abort: Boolean, pause: Boolean, remain: Int, timeout: Option[Int]): TrySeq[ObserveCommand.Result] =
+    if(remain < tic) {
+      Log.info(s"Simulate $name observation completed")
+      TrySeq(ObserveCommand.Success)
+    } else if(stop) TrySeq(ObserveCommand.Stopped)
+      else if(abort) TrySeq(ObserveCommand.Aborted)
+      else if(pause) TrySeq(ObserveCommand.Paused)
+      else if(timeout.exists(_<= 0)) TrySeq.fail(SeqexecException(new TimeoutException()))
+      else {
+        Thread.sleep(tic.toLong)
+        observeTic(stopFlag.get, abortFlag.get, pauseFlag.get, remain - tic, timeout.map(_ - tic))
+      }
+
+  def observe(obsid: ImageFileId, expTime: Time): SeqAction[ObserveCommand.Result] = EitherT( Task {
+    Log.info(s"Simulate taking $name observation with label $obsid")
+    pauseFlag.set(false)
+    stopFlag.set(false)
+    abortFlag.set(false)
+    observeTic(false, false, false, expTime.millis.toInt,
+      (expTime.value > 0.0).option(expTime.toMilliseconds.toInt + 2 * tic))
+  } )
+
+  def applyConfig[C: Show](config: C): SeqAction[Unit] = EitherT( Task {
+    Log.info(s"Simulate applying $name configuration ${config.shows}")
+    Thread.sleep(1000)
```

Comment:
As an example of my previous comment. 1s is sometimes too short for development. And other methodse, e.g. `stopObserve` have no delay